### DISCUSSION TITLE
Query on Federated learning

### DISCUSSION THREAD

**Discussion by DSRajesh** [Project-MONAI/MONAI#7113] [Q&A]
Hello 
 Does monai support Federated Learning for RetinaNet detector ?    Suppose we have a situation like the following: 

Data is distributed across geographical locations. Compute nodes are co located with data.
 The issue is simply that the data exists on different geographical locations in different servers, and so does the training hardware.
 
What is the impact on accuracy and training time ?

Thanks

Rajesh

**Comment by KumoLiu** [comment]
Hi @DSRajesh, thanks for your interest here.

IMO, federated learning is always possible and won't be made difficult by your task.
We have a simple example of segmentation using NVFlare, it's located [here](https://github.com/Project-MONAI/tutorials/tree/main/federated_learning/nvflare).

Hope it can help.
Thanks!

**Reply by DSRajesh** [reply]
Thanks KumoLiu

I had done a code execution walk through of 
https://github.com/NVIDIA/NVFlare/tree/main/examples/advanced/cifar10/cifar10-real-world

I became curious  to know how  to repeat these experiments if I assume the dataset is distributed at say 8 different machines at different locations globally. Are there some .Yaml   or .json files which can specify the location of these systems, say using hostname port address combinations or such things   ?   I could'nt find such things in this example.   Also assume that I found such a mention in  a    .json   file of :  https://github.com/NVIDIA/NVFlare/blob/dev/integration/monai/examples/spleen_ct_segmentation_local/
in its   ..../job/app/config/spleen_ct_segmentation/configs/multi_gpu_train.json    .   It would be great if you can help me in this regard.

Thanking you
Rajesh

**Reply by DSRajesh** [reply]
Also, is there a NVFlare example for object detection Task ?

**Reply by KumoLiu** [reply]
Hi @DSRajesh, do you mean this YAML file?
https://github.com/NVIDIA/NVFlare/blob/dev/integration/monai/examples/spleen_ct_segmentation_local/workspaces/secure_project.yml
https://nvflare.readthedocs.io/en/latest/programming_guide/provisioning_system.html

For now, we don't have an example for detection, but we have [a detection model](https://github.com/Project-MONAI/model-zoo/tree/dev/models/lung_nodule_ct_detection) in Model Zoo, perhaps you could refer to the steps in https://github.com/NVIDIA/NVFlare/tree/dev/integration/monai/examples/spleen_ct_segmentation_local.

Thanks!
